torch/_refs/__init__.py

```diff
@@ -704,6 +713,24 @@ def sub(
     aten_op=None,  # CompositeImplicitAutograd
 )
 
+
+def _xlogy(a: Union[Tensor, NumberType], b: Union[Tensor, NumberType]):
+    if isinstance(a, Tensor) and isinstance(b, Number):
+        b = prims._wrap_scalar(b, dtype=a.dtype, device=a.device)
+    elif isinstance(b, Tensor) and isinstance(a, Number):
+        a = prims._wrap_scalar(a, dtype=b.dtype, device=b.device)
+
+    rhs = where(eq(a, 0), a, mul(a, log(b)))
+    return where(isnan(b), b, rhs)
```

Comment:
Cool!

As with the above, would `where(isnan(b), float('nan'), rhs)` be easier for trace executors?

Follow-up question -- can our operators consume `float('nan')`?

Reply:
`float('nan')` works.